### DISCUSSION TITLE
Mirth connect / nextgenhealthcare - response with json object to the sender ?

### DISCUSSION THREAD

**Discussion by yassersaber-ontology** [nextgenhealthcare/connect#4966] [Q&A]
Dears , i am using old mirth 3.2.2 i am trying to do the following so please advise if can be done in same version or in higher version or cannot done at all 

1-Receive HL7 ,message from sender
2-map the values and using some mapped values to query my external database 
3- return the database query result as a response object to the sender again so sender can receive the response and parse it

I try it between two channels and did the part of querying the database but i can save the result in javascript parameter only so any way to save in json object or hl7 and return this json object as response in the same hl7 messages received from the sender 

fyi, i found that i have to change response inpound data type to (raw) in the mirth channel summary data type selection so object can be returned back to the sender channel but when change to hl7 or xml the response not appear in source channel (so this means i cannot return hl7 object or json object ?) in additon i see the returned object not the data itself

**Comment by jonbartels** [comment]
MC 3.2.2 is quite old, upgrading would be a good idea.

Yes this should be possible in MC, even in 3.2.

You should implement:

- A source transformer that receives HL7
- A destination which writes to your database
- A destination which queries data from your database and puts this data into the channelMap or responseMap
- In the destination, you want to write a Response Transformer which turns your query result into an HL7 response and puts that response into the responseMap
- In your source, you want to set "Respond From" to the map value that contains the HL7 response

**Reply by pacmano1** [reply]
As I mentioned over on the forums where you also posted this  - your db query needed to generate the response may need to return more than one row from your database (for example to build repeating segments), using your DB's native ability to return either XML or JSON will make that easier.   It would be helpful perhaps for you to post some code and screenshots.    If posting code, please do consider reading https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
